modules/utxo-lib/src/bitgo/wallet/Psbt.ts

```diff
@@ -175,70 +224,112 @@ function classifyScriptType(input: PsbtInput): ScriptType2Of3 | undefined {
+  const input = checkForInput(psbt.data.inputs, inputIndex);
+  if (input.partialSig?.length || input.tapScriptSig?.length) {
+    throw new Error('Invalid PSBT signature state');
+  }
+  const partialSigs = parsePsbtMusig2PartialSigs(psbt, inputIndex);
+  if (!partialSigs) {
+    return { signatures: undefined, participantPublicKeys: undefined };
+  }
+  const signatures = partialSigs.map((pSig) => pSig.partialSig);
+  const participantPublicKeys = partialSigs.map((pSig) => pSig.participantPubKey);
+  return isTuple<Buffer>(signatures) && isTuple<Buffer>(participantPublicKeys)
+    ? { signatures, participantPublicKeys }
+    : { signatures: [signatures[0]], participantPublicKeys: [participantPublicKeys[0]] };
+}
 
-  if (scriptType === 'p2tr') {
-    if (input.partialSig && input.partialSig.length > 0) {
-      throw new Error('Invalid PSBT signature state');
-    }
-    if (!input.tapScriptSig || input.tapScriptSig.length === 0) {
-      return undefined;
-    }
-    if (input.tapScriptSig.length > 2) {
-      throw new Error('unexpected signature count');
-    }
-    return input.tapScriptSig.length === 1
-      ? [validate(input.tapScriptSig[0].signature)]
-      : [validate(input.tapScriptSig[0].signature), validate(input.tapScriptSig[1].signature)];
+function parseTaprootScriptPathSignatures(psbt: UtxoPsbt, inputIndex: number): Signatures {
+  const input = checkForInput(psbt.data.inputs, inputIndex);
+  if (input.partialSig && input.partialSig.length > 0) {
+    throw new Error('Invalid PSBT signature state');
   }
+  if (!input.tapScriptSig || input.tapScriptSig.length === 0) {
+    return { signatures: undefined };
+  }
+  if (input.tapScriptSig.length > 2) {
+    throw new Error('unexpected signature count');
+  }
+  const signatures = input.tapScriptSig.map((tSig) => tSig.signature);
+  return isTuple<Buffer>(signatures) ? { signatures } : { signatures: [signatures[0]] };
+}
+
+function parseP2msSignatures(psbt: UtxoPsbt, inputIndex: number): Signatures {
+  const input = checkForInput(psbt.data.inputs, inputIndex);
   if (input.tapScriptSig && input.tapScriptSig.length > 0) {
     throw new Error('Invalid PSBT signature state');
   }
   if (!input.partialSig || input.partialSig.length === 0) {
-    return undefined;
+    return { signatures: undefined };
   }
   if (input.partialSig.length > 2) {
     throw new Error('unexpected signature count');
   }
-  return input.partialSig.length === 1
-    ? [validate(input.partialSig[0].signature)]
-    : [validate(input.partialSig[0].signature), validate(input.partialSig[1].signature)];
+  const signatures = input.partialSig.map((pSig) => pSig.signature);
+  return isTuple<Buffer>(signatures) ? { signatures } : { signatures: [signatures[0]] };
+}
+
+function parseSignatures(
+  psbt: UtxoPsbt,
+  inputIndex: number,
+  scriptType: ScriptTypeP2msBased | ScriptTypeTaprootVirtual
+): Signatures | P2trMusig2Signatures {
```

Comment:
reads confusing since `Signatures` should be a supertype of `P2trMusig2Signatures`